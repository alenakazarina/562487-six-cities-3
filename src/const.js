export const BASE_URL = `https://htmlacademy-react-3.appspot.com/six-cities`;

export const SortType = {
  POPULAR: `Popular`,
  PRICE_TO_HIGH: `Price: low to high`,
  PRICE_TO_LOW: `Price: high to low`,
  TOP_RATED: `Top rated first`
};

export const OfferType = {
  house: `House`,
  room: `Private room`,
  apartment: `Apartment`,
  hotel: `Hotel`
};

export const AppRoute = {
  ROOT: `/`,
  OFFER: `/offer/:id`,
  FAVORITES: `/favorites`,
  LOGIN: `/login`
};

export const RATINGS = [
  {
    value: 5,
    title: `perfect`
  },
  {
    value: 4,
    title: `good`
  },
  {
    value: 3,
    title: `not bad`
  },
  {
    value: 2,
    title: `badly`
  },
  {
    value: 1,
    title: `terribly`
  }
];

export const MONTHS = [
  `January`, `February`, `March`, `April`,
  `May`, `June`, `July`, `August`,
  `September`, `October`, `November`, `December`
];

export const AuthStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

export const InitialState = {
  USER: {
    authStatus: `NO_AUTH`,
    user: {
      id: -1,
      email: ``,
      name: ``,
      isPro: false,
      avatarUrl: ``
    }
  },
  FAVORITES: {
    favorites: []
  },
  OFFER: {
    activeOffer: null,
    nearOffers: [],
    comments: []
  },
  OFFERS: {
    offers: [],
    activeCity: ``
  },
  ERRORS: {
    errorStatus: null
  }
};

