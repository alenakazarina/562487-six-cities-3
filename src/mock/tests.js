export const locations = [
  {
    title: `Paris`,
    isActive: true
  },
  {
    title: `Cologne`,
    isActive: false
  },
  {
    title: `Brussels`,
    isActive: false
  },
  {
    title: `Amsterdam`,
    isActive: false
  },
  {
    title: `Hamburg`,
    isActive: false
  },
  {
    title: `Dusseldorf`,
    isActive: false
  }
];

export const offers = [
  {
    id: `01`,
    title: `Beautiful & luxurious apartment at great location`
  },
  {
    id: `02`,
    title: `Wood and stone place`
  },
  {
    id: `03`,
    title: `Canal View Prinsengracht`
  },
  {
    id: `04`,
    title: `Nice, cozy, warm big bed apartment`
  },
  {
    id: `05`,
    title: `Cozy rooms in the city center`
  }
];

export const title = `Canal View Prinsengracht`;

export const onTitleClick = () => {};
