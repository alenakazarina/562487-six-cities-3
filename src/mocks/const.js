export const IMAGE_PATH = `./img/`;

export const MAP_ZOOM = 12;

export const MONTHS = [
  `January`, `February`, `March`, `April`, `May`, `June`, `July`,
  `August`, `September`, `October`, `November`, `December`
];

const OfferTypes = {
  apartment: `Apartment`,
  room: `Private Room`,
  house: `House`,
  hotel: `Hotel`
};

export const OffersData = {
  maxRating: 5,
  types: Object.values(OfferTypes),
  titles: [
    `Canal View room`,
    `Nice, cozy, warm big bed apartment`,
    `Wood and stone place`,
    `Beautiful & luxurious apartment at great location`,
    `Cozy rooms in the city center`
  ],
  images: [
    `apartment-01.jpg`,
    `apartment-02.jpg`,
    `apartment-03.jpg`,
    `apartment-04.jpg`,
    `apartment-05.jpg`,
  ],
  amenities: [`Wi-Fi`, `Washing machine`, `Towels`, `Heating`, `Coffee machine`, `Baby seat`, `Kitchen`, `Cabel TV`, `Fridge`],
  description: `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`
};

export const UserData = {
  names: [`Douglas`, `James`, `Matthew`, `John`, `Leo`, `Robert`, `Dirk`],
  avatarUrls: [
    `user-1.jpg`,
    `user-2.jpg`,
    `user-3.jpg`,
    `user-4.jpg`,
    `user-5.jpg`,
  ],
  comment: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`
};

export const Cities = {
  PARIS: {
    name: `Paris`,
    cityCenter: {
      latitude: 48.859335,
      longitude: 2.350730
    },
    offersLocations: [{
      latitude: 48.872344,
      longitude: 2.360418
    }, {
      latitude: 48.866015,
      longitude: 2.346708
    }, {
      latitude: 48.850586,
      longitude: 2.327006
    }, {
      latitude: 48.855066,
      longitude: 2.316033
    }]
  },
  COLOGNE: {
    name: `Cologne`,
    cityCenter: {
      latitude: 50.939485,
      longitude: 6.948775
    },
    offersLocations: [{
      latitude: 50.937764,
      longitude: 6.961390
    }, {
      latitude: 50.945324,
      longitude: 6.960608
    }, {
      latitude: 50.933751,
      longitude: 6.936701
    }, {
      latitude: 50.931175,
      longitude: 6.948593
    }]
  },
  BRUSSELS: {
    name: `Brussels`,
    cityCenter: {
      latitude: 50.855889,
      longitude: 4.351608
    },
    offersLocations: [{
      latitude: 50.854885,
      longitude: 4.345541
    }, {
      latitude: 50.858440,
      longitude: 4.329629
    }, {
      latitude: 50.835941,
      longitude: 4.346313
    }, {
      latitude: 50.834979,
      longitude: 4.351570
    }]
  },
  AMSTERDAM: {
    name: `Amsterdam`,
    cityCenter: {
      latitude: 52.371512,
      longitude: 4.893750
    },
    offersLocations: [{
      latitude: 52.370955,
      longitude: 4.873096
    }, {
      latitude: 52.379553,
      longitude: 4.873096
    }, {
      latitude: 52.390955,
      longitude: 4.909309
    }, {
      latitude: 52.360955,
      longitude: 4.909309
    }]
  },
  HAMBURG: {
    name: `Hamburg`,
    cityCenter: {
      latitude: 53.556238,
      longitude: 9.985139
    },
    offersLocations: [{
      latitude: 53.553348,
      longitude: 9.971704
    }, {
      latitude: 53.552588,
      longitude: 10.019478
    }, {
      latitude: 53.561941,
      longitude: 10.018891
    }, {
      latitude: 53.548758,
      longitude: 9.978013
    }]
  },
  DUSSELDORF: {
    name: `Dusseldorf`,
    cityCenter: {
      latitude: 51.225322,
      longitude: 6.774754
    },
    offersLocations: [{
      latitude: 51.219590,
      longitude: 6.774384
    }, {
      latitude: 51.232822,
      longitude: 6.780301
    }, {
      latitude: 51.220024,
      longitude: 6.792541
    }, {
      latitude: 51.215356,
      longitude: 6.781649
    }]
  }
};
