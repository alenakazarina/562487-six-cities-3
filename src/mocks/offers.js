import {OffersData} from './const';
import {
  generateId,
  getImage,
  getRandomBoolean,
  getRandomNumber,
  getRandomFromArray,
} from './utils';

const generateOffer = () => {
  return {
    id: generateId(),
    previewImage: getImage(OffersData.images),
    title: getRandomFromArray(OffersData.titles),
    isFavorite: getRandomBoolean(),
    isPremium: getRandomBoolean(),
    rating: getRandomNumber({min: 0, max: OffersData.maxRating}),
    type: getRandomFromArray(OffersData.types),
    price: getRandomNumber({min: 0, max: 1000})
  };
};

export const generateOffers = (count = 3) => {
  return new Array(count).fill(``).map(generateOffer);
};

export const offers = [
  {
    city: `Paris`,
    offers: generateOffers(4)
  },
  {
    city: `Cologne`,
    offers: generateOffers(5)
  },
  {
    city: `Brussels`,
    offers: generateOffers(4)
  },
  {
    city: `Amsterdam`,
    offers: generateOffers(5)
  },
  {
    city: `Hamburg`,
    offers: generateOffers(4)
  },
  {
    city: `Dusseldorf`,
    offers: generateOffers(5)
  }
];
