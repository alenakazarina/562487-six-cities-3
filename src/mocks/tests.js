import configureStore from 'redux-mock-store';
import {AuthStatus} from '../reducers/user/user';
import {cityOffers, reviews, appUsers} from './const';

const defaultUser = {
  id: -1,
  email: ``,
  name: ``,
  isPro: false,
  avatarUrl: ``
};

const mockStore = configureStore([]);

export const storeWithAuth = mockStore({
  USER: {
    authStatus: AuthStatus.AUTH,
    user: appUsers[0]
  },
  OFFERS: {
    offers: cityOffers,
    activeCity: `Paris`
  },
  OFFER: {
    pageOffer: cityOffers[1],
    activeOffer: cityOffers[2],
    nearOffers: cityOffers.slice(0, 2),
    comments: reviews
  },
  FAVORITES: {
    favorites: cityOffers
  }
});

export const storeWithNoAuth = mockStore({
  USER: {
    authStatus: AuthStatus.NO_AUTH,
    user: defaultUser
  },
  OFFERS: {
    offers: cityOffers,
    activeCity: `Paris`
  },
  OFFER: {
    pageOffer: null,
    activeOffer: null,
    nearOffers: [],
    comments: []
  },
  FAVORITES: {
    favorites: []
  }
});

export const storeWithNoOffers = mockStore({
  USER: {
    authStatus: AuthStatus.NO_AUTH,
    user: defaultUser
  },
  OFFERS: {
    offers: [],
    activeCity: `Paris`
  },
  OFFER: {
    pageOffer: null,
    activeOffer: null,
    nearOffers: [],
    comments: []
  },
  FAVORITES: {
    favorites: []
  }
});
