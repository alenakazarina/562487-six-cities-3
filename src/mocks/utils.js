import nanoid from 'nanoid';
import {IMAGE_PATH, OffersData} from './const';

const getRandomNumber = ({min = 0, max = 1}) => {
  return min + Math.floor(max * Math.random());
};

const getRandomFromObject = (obj = {}) => {
  const randomKey = getRandomFromArray(Object.keys(obj));
  return obj[randomKey];
};

const getRandomFromArray = (arr = []) => {
  const randomIndex = getRandomNumber({min: 0, max: arr.length});
  return arr[randomIndex];
};

const getRandomArray = ({items = [], minLength = 0, maxLength = 1}) => {
  const length = getRandomNumber({min: minLength, max: maxLength + 1});
  return new Array(length).fill(``).map(() => {
    const index = getRandomNumber(0, items.length);
    return items[index];
  });
};

const getRandomBoolean = () => {
  return Math.random() > 0.5;
};

const generateId = () => {
  return nanoid();
};

const getImage = (items = []) => {
  const randomIndex = getRandomNumber({min: 0, max: items.length});
  return IMAGE_PATH.concat(items[randomIndex]);
};

const generateImages = () => {
  return OffersData.images.map((image) => IMAGE_PATH.concat(image));
};

const formatMonthYear = (date = new Date()) => {
  return `${date.getMonth() + 1} ${date.getFullYear()}`;
};

export {
  getRandomNumber,
  getRandomArray,
  getRandomBoolean,
  generateId,
  getImage,
  generateImages,
  getRandomFromArray,
  getRandomFromObject,
  formatMonthYear
};
