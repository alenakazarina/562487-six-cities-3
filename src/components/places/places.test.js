import React from 'react';
import renderer from 'react-test-renderer';
import Places from './places';
import {cityOffers} from '../../mocks/tests';
import {SortType} from '../../const';

const mockFn = () => {};

describe(`Places`, () => {
  it(`should render Places correctly`, () => {
    const tree = renderer.create(
        <Places
          offers={cityOffers}
          activeSortType={SortType.POPULAR}
          onTitleClick={mockFn}
          onSortTypeChange={mockFn}
          onCardHoverChange={mockFn}
        />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
