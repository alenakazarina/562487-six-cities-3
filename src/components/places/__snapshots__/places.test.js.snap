// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Places should render Places correctly 1`] = `
<section
  className="cities__places places"
>
  <h2
    className="visually-hidden"
  >
    Places
  </h2>
  <b
    className="places__found"
  >
    2 places
     to stay in 
    Amsterdam
  </b>
  <form
    action="#"
    className="places__sorting"
    method="get"
  >
    <span
      className="places__sorting-caption"
    >
      Sort by 
    </span>
    <span
      className="places__sorting-type"
      onClick={[Function]}
      tabIndex="0"
    >
      Popular
      <svg
        className="places__sorting-arrow"
        height="4"
        width="7"
      >
        <use
          xlinkHref="#icon-arrow-select"
        />
      </svg>
    </span>
    <ul
      className="places__options places__options--custom "
      onClick={[Function]}
    >
      <li
        className="places__option places__option--active"
        onClick={[Function]}
        tabIndex="0"
      >
        Popular
      </li>
      <li
        className="places__option "
        onClick={[Function]}
        tabIndex="0"
      >
        Price: low to high
      </li>
      <li
        className="places__option "
        onClick={[Function]}
        tabIndex="0"
      >
        Price: high to low
      </li>
      <li
        className="places__option "
        onClick={[Function]}
        tabIndex="0"
      >
        Top rated first
      </li>
    </ul>
  </form>
  <div
    className="cities__places-list places__list tabs__content"
    onMouseLeave={[Function]}
  >
    <article
      className="cities__place-card place-card"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      
      <div
        className="cities__image-wrapper place-card__image-wrapper"
      >
        <a
          href="#"
        >
          <img
            alt="Place image"
            className="place-card__image"
            height="200"
            src="image"
            width="260"
          />
        </a>
      </div>
      <div
        className="place-card__info"
      >
        <div
          className="place-card__price-wrapper"
        >
          <div
            className="place-card__price"
          >
            <b
              className="place-card__price-value"
            >
              €
              200
            </b>
            <span
              className="place-card__price-text"
              style={
                Object {
                  "marginLeft": "3px",
                }
              }
            >
              / night
            </span>
          </div>
          <button
            className="place-card__bookmark-button button place-card__bookmark-button--active"
            type="button"
          >
            <svg
              className="place-card__bookmark-icon"
              height={19}
              width={18}
            >
              <use
                xlinkHref="#icon-bookmark"
              />
            </svg>
            <span
              className="visually-hidden"
            >
              To bookmarks
            </span>
          </button>
        </div>
        <div
          className="place-card__rating rating"
        >
          <div
            className="place-card__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "60%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <h2
          className="place-card__name"
        >
          <a
            onClick={[Function]}
          >
            title
          </a>
        </h2>
        <p
          className="place-card__type"
        />
      </div>
    </article>
    <article
      className="cities__place-card place-card"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      
      <div
        className="cities__image-wrapper place-card__image-wrapper"
      >
        <a
          href="#"
        >
          <img
            alt="Place image"
            className="place-card__image"
            height="200"
            src="image"
            width="260"
          />
        </a>
      </div>
      <div
        className="place-card__info"
      >
        <div
          className="place-card__price-wrapper"
        >
          <div
            className="place-card__price"
          >
            <b
              className="place-card__price-value"
            >
              €
              500
            </b>
            <span
              className="place-card__price-text"
              style={
                Object {
                  "marginLeft": "3px",
                }
              }
            >
              / night
            </span>
          </div>
          <button
            className="place-card__bookmark-button button place-card__bookmark-button--active"
            type="button"
          >
            <svg
              className="place-card__bookmark-icon"
              height={19}
              width={18}
            >
              <use
                xlinkHref="#icon-bookmark"
              />
            </svg>
            <span
              className="visually-hidden"
            >
              To bookmarks
            </span>
          </button>
        </div>
        <div
          className="place-card__rating rating"
        >
          <div
            className="place-card__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "80%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <h2
          className="place-card__name"
        >
          <a
            onClick={[Function]}
          >
            title
          </a>
        </h2>
        <p
          className="place-card__type"
        />
      </div>
    </article>
  </div>
</section>
`;
